docs/development/best-practices.md

```diff
@@ -115,6 +115,10 @@ try {
 }
 ```
 
+### Aysnc functions
+
+Never use `reject` in async functions, instead throw an error class type.
```

Comment:
The lines of code that @viceice commented on were:

```typescript
    return Promise.reject({
      ...e,
      from,
      to,
    });
```

Should the documentation tell people to never use `Promise.reject` or just `reject`? And do we want to create an example of the correct code?

Reply:
```suggestion
Never use `Promise.resolve` in async functions.
Never use `Promise.reject` in async functions, instead throw an `Error` class type.
```